client/me/purchases/list/item/index.jsx

```diff
@@ -39,7 +39,7 @@ const PurchaseItem = React.createClass( {
 			);
 		}
 
-		if ( isRenewing( purchase ) ) {
+		if ( isRenewing( purchase ) && purchase.renewMoment ) {
```

Comment:
If the site upgrades endpoint is returning incomplete `purchases` objects, I think we should update it. If `purchase.renew_date` isn't set on purchases that renew, the purchase will appear to be expiring here even if it is renewing.